src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -20,6 +21,8 @@ internal static class AppHost
         private const string AppBinaryPathPlaceholder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2";
         private readonly static byte[] AppBinaryPathPlaceholderSearchValue = Encoding.UTF8.GetBytes(AppBinaryPathPlaceholder);
 
+        public const int DefaultRetryDelayMilliseconds = 1000;
```

Comment:
This file is getting removed as the app host generation code has been consolidated in core-setup to prevent three copies: core-setup, cli, sdk.

https://github.com/dotnet/sdk/pull/3447

My suggestion for this would be to bump the retry up to the task caller of Create. We could also PR this to core-setup, but it feels like retrying is more of a build thing than an app host library thing.

Reply:
Code looks good, we just have to find a way to merge with #3447

Reply:
cc @wli3

Reply:
this change is conflicting

Reply:
Will and I talked this over.  I'm going to move the retry to the outer `HostWriter` call to retry the whole operation again.  It'll be slightly less efficient in the error case, but won't require us to modify `HostWriter` in ask mode.